From mbeattie@sable.ox.ac.uk Tue Jun 24 21:39:43 1997
From: mbeattie@sable.ox.ac.uk (Malcolm Beattie)
Newsgroups: comp.lang.perl.misc
Subject: Re: Perl Compiler Trubles
Date: 16 Jun 1997 12:00:41 GMT
Organization: Oxford University, England
Lines: 85
Message-ID: <5o39t9$4ko@news.ox.ac.uk>
References: <33A50BA1.6E2@chim1.unifi.it>
NNTP-Posting-Host: sable.ox.ac.uk
Xref: ddsw1 comp.lang.perl.misc:81206

In article <33A50BA1.6E2@chim1.unifi.it>,
Massimo Fontanelli  <maxi@chim1.unifi.it> wrote:
>I have a Perl Compiler Toolkit problem, so I posted a message, but many
>informations
>about were missing, let's try again?
>
>My environment is: Dec Aplha running DEC Unix V3.2, Perl Version 5.004 +
                                                                  ^^^^^

There's the problem: 5.004 introduced a new keyword which means the
compiler kit's arrays of optypes and names are incorrect from a
certain point onwards.

>Perl Compiler Toolkit,
>both was compiled without problems using DEC OSF/1 C compiler
>
>The origin of the truble was that I wish to compile the following code:
>
>    #!/usr/bin/perl
>    $the_dir="/tmp";
>    opendir(CDIR, $the_dir);
>    @files = grep(/./,readdir(CDIR));
>    closedir(CDIR);
>    print "\n@files\n";
>
>The message of the toolkit is:
>>test syntax OK
>>Can't locate object method "first" via package "B::OP" at /usr/local/lib/perl5/s
>>ite_perl/B.pm line 95.

Here's a patch for that problem. Now that 5.004 is out and the current
compiler snapshot is being tested, I should be able to release alpha4
very soon. For now, try this:

------------------------------ cut here ------------------------------
Index: perlext/Compiler/ccop.c
==== //depot/perlext/Compiler/ccop.c#1 - //depot/perlext/Compiler/ccop.c#2 ====
*** perlext/Compiler/ccop.c.old	Mon Jun 16 11:57:04 1997
--- perlext/Compiler/ccop.c	Mon Jun 16 11:57:04 1997
***************
*** 404,409 ****
--- 404,412 ----
  	cc_listop,		/* prtf */
  	cc_listop,		/* print */
  	cc_listop,		/* sysopen */
+ #if PATCHLEVEL > 3
+ 	cc_listop,		/* sysseek */
+ #endif
  	cc_listop,		/* sysread */
  	cc_listop,		/* syswrite */
  	cc_listop,		/* send */
Index: perlext/Compiler/ccop.h
==== //depot/perlext/Compiler/ccop.h#1 - //depot/perlext/Compiler/ccop.h#2 ====
*** perlext/Compiler/ccop.h.old	Mon Jun 16 11:57:12 1997
--- perlext/Compiler/ccop.h	Mon Jun 16 11:57:12 1997
***************
*** 1,3 ****
--- 1,7 ----
+ #ifndef PATCHLEVEL
+ #include "patchlevel.h"
+ #endif
+ 
  typedef enum {
      OPc_NULL,	/* 0 */
      OPc_BASEOP,	/* 1 */
***************
*** 230,235 ****
--- 234,242 ----
  	"pp_prtf",
  	"pp_print",
  	"pp_sysopen",
+ #if PATCHLEVEL > 3
+ 	"pp_sysseek",
+ #endif
  	"pp_sysread",
  	"pp_syswrite",
  	"pp_send",
------------------------------ cut here ------------------------------

--Malcolm

-- 
Malcolm Beattie <mbeattie@sable.ox.ac.uk>
Oxford University Computing Services
"Widget. It's got a widget. A lovely widget. A widget it has got." --Jack Dee


